src/controller/CHIPDeviceController.h

```diff
@@ -468,9 +469,15 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
      *         in the Init() call.
      *
      * @param[in] remoteDeviceId        The remote device Id.
-     * @param[in] params                The Rendezvous connection parameters
+     * @param[in] rendezvousParams      The Rendezvous connection parameters
+     * @param[in] commssioningParams    The commissioning parameters (uses defualt if not supplied)
      */
-    CHIP_ERROR PairDevice(NodeId remoteDeviceId, RendezvousParameters & params);
+    CHIP_ERROR PairDevice(NodeId remoteDeviceId, RendezvousParameters & rendezvousParams);
+    CHIP_ERROR PairDevice(NodeId remoteDeviceId, RendezvousParameters & rendezvousParams,
+                          CommissioningParameters & commissioningParams);
+
+    CHIP_ERROR EstablishPASEConnection(NodeId remoteDeviceId, RendezvousParameters & params);
```

Comment:
I believe the existing setup was at least partly to prevent SDK consumers from using PASE for something that is not commissioning....

Ideally @pan-apple would take a look at this, but he (like everyone at Apple) is off this next week.  I would appreciate it if this PR waits until he gets back, if at all possible.

If the real problem here is that our existing commissioning flow is too opinionated for some consumers, maybe we should have a conversation about how to address that without opening up the floodgates for arbitrary operation over PASE....

Reply:
Current BLE flow doesn't even force you to resolve the device before performing operations, so I don't think the SDK as it stands is taking a particularly strong stance on not sending over PASE. In fact, until this pr goes in https://github.com/project-chip/connectedhomeip/pull/11993, we silently fall back to PASE when getting the device. 

The spec doesn't take a strong stance on what operations can and can't happen during commissioning. Especially with the custom commissioning flows, it's hard to say for sure what can and can't take place over a PASE connection, and just not providing the code isn't a strong deterrent.

If there are rules about what operations can and can't happen over PASE, we should enforce those. We should also enforce other security things like having the failsafe armed for PASE operations. Note that those things are also not testable unless we have a separate PASE establishment function.

Reply:
@bzbarsky-apple I agree that PASE should only be used for commisioning, however code should be able to have sufficient abstractions as well.

Allowing PASE only during commisioning should be enforced in some different way than denying code reuse.

Reply:
> I believe the existing setup was at least partly to prevent SDK consumers from using PASE for something that is not commissioning....
> If the real problem here is that our existing commissioning flow is too opinionated for some consumers, maybe we should have a conversation about how to address that without opening up the floodgates for arbitrary operation over PASE....

We are not opening "arbitrary operation over PASE". I see this PR as making things more orthogonal and cleaner. 

There is nothing in the spec that I can see that limits what you can do over PASE *during commissioning*. There is no code in the PR that I can see that further or additionally allows arbitrary PASE session establishments at runtime outside of commissioning windows.

Overall, since this PR doesn't attempt to "open floodgates", I would keep to only looking at the technical aspects. This PR is not trying to allow arbitrary runtime PASE, just clean-up the commisisoning flow to be cleaner.

Spec:

----8<----8<----8<----8<----8<
> . Security setup with PASE (see <<ref_PASE>>): Establish encryption keys between the Commissioner and Commissionee using PASE. All CHIP messages exchanged between the Commissioner and Commissionee are encrypted using these PASE-derived keys. The process also establishes an attestation challenge used during the <<ref_Device_Attestation_Procedure,device attestation procedure>>.

> A commissioner can reconfigure the Commissionee multiple times over the operational network after the commissioning is complete or over the commissioning channel after PASE-derived encryption keys are established during commissioning. The commissioning flows are described in <<ref_Commissioning_Flows>>.

> During commissioning and unicast communication, a <<ref_Discovery, discovery>> mechanism is provided to determine peer IPv6 addresses and operational parameters. <<ref_SessionEstablishment, Secure session establishment>> mechanisms are provided using either <<ref_CASE, certificates (CASE)>> or <<ref_PASE, shared passcodes (PASE)>>. 

> All communication between Nodes, be it unicast or groupcast, is protected with a <<ref_MessageSecurity, secure messaging layer>>. Every message besides the <<ref_Discovery, discovery>>, <<ref_CASE, CASE establishment>>, and <<ref_PASE, PASE establishment>> messages is encrypted by this secure messaging layer.

> <<ref_PASE, PASE>> SHALL be used as a session establishment mechanism during device commissioning. <<ref_PASE, PASE>> SHALL NOT be used as a session establishment mechanism for any other session.

> Unless otherwise specified, the <<ref_CASE, CASE>>, <<ref_PASE, PASE>>, <<User_Directed_Commissioning,User-Directed Commissioning>> protocol, and <<ref_SC_StatusReports, Secure Channel Status Report>> messages SHALL be the only allowed *unencrypted* messages.

----8<----8<----8<----8<----8<

Reply:
Regardless of what we expose as external API, we should agree to have clean, modular APIs internal to the SDK to keep PASE and Commissioning distinct and separate layers to drive modularity, testability, and code maintenance.  This PR seems a step in the right direction to that regard.